quantum/keycode_string.c

```diff
@@ -0,0 +1,481 @@
+            append_unary_keycode(PSTR("DF"), number_string(QK_DEF_LAYER_GET_LAYER(keycode), 10));
+            return;
+
+        case QK_TOGGLE_LAYER ... QK_TOGGLE_LAYER_MAX:  // TG(layer) key.
+            append_unary_keycode(PSTR("TG"), number_string(QK_TOGGLE_LAYER_GET_LAYER(keycode), 10));
+            return;
+
+        case QK_ONE_SHOT_LAYER ... QK_ONE_SHOT_LAYER_MAX:  // OSL(layer) key.
+            append_unary_keycode(PSTR("OSL"), number_string(QK_ONE_SHOT_LAYER_GET_LAYER(keycode), 10));
+            return;
+
+        case QK_LAYER_TAP_TOGGLE ... QK_LAYER_TAP_TOGGLE_MAX:  // TT(layer) key.
+            append_unary_keycode(PSTR("TT"), number_string(QK_LAYER_TAP_TOGGLE_GET_LAYER(keycode), 10));
+            return;
+
+        case QK_PERSISTENT_DEF_LAYER ... QK_PERSISTENT_DEF_LAYER_MAX:  // PDF(layer) key.
+            append_unary_keycode(PSTR("PDF"), number_string(QK_PERSISTENT_DEF_LAYER_GET_LAYER(keycode), 10));
+            return;
+
+        // Mod-tap MT(mod,kc) key. This implementation formats the MT keys where
+        // one modifier is applied. For MT keys with multiple modifiers, the mod
+        // arg is written numerically as a hex code.
+        case QK_MOD_TAP ... QK_MOD_TAP_MAX: {
+            uint8_t mods = QK_MOD_TAP_GET_MODS(keycode);
+            const bool is_rhs = mods > 15;
+            mods &= 15;
+            if (mods != 0 && (mods & (mods - 1)) == 0) {  // One mod is set.
+                append_char(is_rhs ? 'R' : 'L');
+                append_P(&mod_names[4 * biton(mods)]);
+                append_P(PSTR("_T("));
+            } else {
+                append_P(PSTR("MT("));
+                append_number(mods, 16);
+                append_char(',');
+            }
+            append_keycode(QK_MOD_TAP_GET_TAP_KEYCODE(keycode));
+            append_char(')');
+        }   return;
+
+        case QK_TAP_DANCE ... QK_TAP_DANCE_MAX:  // Tap dance TD(i) key.
+            append_unary_keycode(PSTR("TD"), number_string(QK_TAP_DANCE_GET_INDEX(keycode), 10));
+            return;
+
+#ifdef UNICODE_ENABLE
+        case QK_UNICODE ... QK_UNICODE_MAX:  // Unicode UC(codepoint) key.
+            append_unary_keycode(PSTR("UC"), number_string(QK_UNICODE_GET_CODE_POINT(keycode), 16));
+            return;
+#elif defined(UNICODEMAP_ENABLE)
+        case QK_UNICODEMAP ... QK_UNICODEMAP_MAX:  // Unicode Map UM(i) key.
+            append_unary_keycode(PSTR("UM"), number_string(QK_UNICODEMAP_GET_INDEX(keycode), 10));
+            return;
+
+        case QK_UNICODEMAP_PAIR ... QK_UNICODEMAP_PAIR_MAX: {  // UP(i,j) key.
+            const uint8_t i = QK_UNICODEMAP_PAIR_GET_UNSHIFTED_INDEX(keycode);
+            const uint8_t j = QK_UNICODEMAP_PAIR_GET_SHIFTED_INDEX(keycode);
+            append_P(PSTR("UP("));
+            append_number(i, 10);
+            append_char(',');
+            append_number(j, 10);
+            append_char(')');
+        }   return dest;
```

Comment:
```suggestion
            append_char(')');
            return;
        }
```

Looks like a leftover from a previous iteration? :)

Reply:
Thanks for that! I'll fix this in the next commit.